### PR TITLE
Unified more dependencies across the tree

### PR DESCRIPTION
**Pull request by davkean**
Unified VisualStudio, Moq and System.Composition dependencies across the tree

### CODE REVIEW

**Comment by davkean** [comment]
tag @jasonmalinowski @tmat @agocke

**Comment by davkean** [comment]
Make note that due to https://github.com/NuGet/Home/issues/2558, we can't name the projects the same as their nuget packages.

File: src/Dependencies/VisualStudioText/project.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "supports": { },
+  "dependencies": {
+    "Microsoft.VisualStudio.Text.Data": "14.1.24720",
+    "Microsoft.VisualStudio.Text.Logic": "14.1.24720",
+    "Microsoft.VisualStudio.Text.UI": "14.1.24720",
+    "Microsoft.VisualStudio.Text.UI.Wpf": "14.1.24720"
```
Comment (jasonmalinowski) [inline src/Dependencies/VisualStudioText/project.json]: Should we just merge this with your VisualStudioEditor dependency? I admit that if I was asked which of these two a new reference should go into, I wouldn't have a clear answer.

File: src/InteractiveWindow/Editor/InteractiveWindow.csproj [hunk 2]
```diff
@@ -124,6 +124,16 @@
     <PublicAPI Include="PublicAPI.Unshipped.txt" />
     <PublicAPI Include="PublicAPI.Shipped.txt" />
   </ItemGroup>
+  <ItemGroup>
+    <ProjectReference Include="..\..\Dependencies\VisualStudioEditor\VisualStudioEditor.csproj">
```
Comment (jasonmalinowski) [inline src/InteractiveWindow/Editor/InteractiveWindow.csproj]: If we decide my other comment about merging these is daft, shouldn't VisualStudioEditor depend on VisualStudioText? Can you actually consume one without the other?

**Comment by jasonmalinowski** [comment]
@dotnet/roslyn-ide should also take a look at this, since they'll need to understand what this means going forward.

File: src/Dependencies/VisualStudioText/project.json (hunk 1 above)
Comment (davkean) [inline src/Dependencies/VisualStudioText/project.json]: Apparently this was split by design - to avoid loading Editor binaries in CodeLens.

Comment (davkean) [inline src/Dependencies/VisualStudioText/project.json]: And hence, if so, we should enforce it.

**Comment by davkean** [comment]
ping @dotnet/roslyn-ide @jasonmalinowski @tmat @agocke

File: src/InteractiveWindow/Editor/InteractiveWindow.csproj (hunk 2 above)
Comment (davkean) [inline src/InteractiveWindow/Editor/InteractiveWindow.csproj]: Yes, that's exactly what features does.

File: src/Dependencies/Composition/project.json [hunk 3]
```diff
@@ -0,0 +1,9 @@
+{
+  "supports": {},
+  "dependencies": {
+    "Microsoft.Composition": "1.0.27"
```
Comment (Pilchie) [inline src/Dependencies/Composition/project.json]: What is the advantage of this approach for single dependency projects like this?  I get it for the VS binaries, but not Moq and Composition.

**Comment by Pilchie** [comment]
:+1:

File: src/Dependencies/Composition/project.json (hunk 3 above)
Comment (tmat) [inline src/Dependencies/Composition/project.json]: So that we have the version in one file vs in many. Updating to a new version is easier.

**Comment by jmarolf** [comment]
test vsi please

File: src/Dependencies/Composition/project.json (hunk 3 above)
Comment (davkean) [inline src/Dependencies/Composition/project.json]: We've already run into situations where versions get updated in master, but projects in future get missed. Now, you update in once place and when it moves to future all places get updated.

**Comment by davkean** [comment]
@jmarolf You like to see the world burn don't you?

**Comment by jmarolf** [comment]
@davkean why not test when you could test?

**Comment by davkean** [comment]
:) Good catch - I should have done that.

**Comment by davkean** [comment]
@dotnet-bot test vsi please

**Pull request merged**